dapps/frenemies/src/routes/Setup.tsx

```diff
@@ -32,61 +33,93 @@ export function Setup() {
   );
   const queryClient = useQueryClient();
 
+  useEffect(() => {
+    if (
+      isSuccess &&
+      legacyScorecard.isSuccess &&
+      !scorecard &&
+      legacyScorecard.data
+    ) {
+      navigate("/migrate", { replace: true });
+    }
+  }, [isSuccess, scorecard, legacyScorecard]);
+
+  useEffect(() => {
```

Comment:
Maybe a dumb question, but does ordering matter here? i.e. should we check for the account before the scorecard(s) so that we redirect to `/connect` before `/migrate`?

Reply:
It shouldn't, because these should are all technically mutually exclusive states.

Reply:
Though I agree it's messy how this is written, ideally would not have as much of this inline and imperative like this